src/components/ImageEditor.vue

```diff
@@ -125,6 +130,39 @@ export default {
 				logger.error('Error saving image', { error })
 			}
 		},
+
+		/**
+		 * Show warning if unsaved changes
+		 */
+		onExitWithoutSaving() {
+			OC.dialogs.confirmDestructive(
+				editorTranslations.changesLoseConfirmation + '\n\n' + editorTranslations.changesLoseConfirmationHint,
+				t('viewer', 'Unsaved changes'),
+				{
+					type: OC.dialogs.YES_NO_BUTTONS,
+					confirm: t('viewer', 'Drop changes'),
+					confirmClasses: 'error',
+					cancel: editorTranslations.cancel,
+				},
+				(decision) => {
+					if (!decision) {
+						return
+					}
+					this.onClose('warning-ignored', false)
+				}
+			)
+		},
+
+		// Key Handlers, override default Viewer arrow and escape key
+		handleKeydown(event) {
+			event.stopImmediatePropagation()
+			// escape key
+			if (event.keyCode === 27) {
```

Comment:
rather use this?
https://github.com/nextcloud/calendar/blob/39709a9544efd0aed95a72a367493a2f38b46357/src/mixins/EditorMixin.js#L433

Reply:
Right! keyCode is even deprecated
I was taking it from https://github.com/nextcloud/nextcloud-vue/blob/94e9e958fa5f3d343660bafc10f1166e2dba9b8c/src/components/Modal/Modal.vue#L598-L611, which we should fix too :)